ql/cashflows/iborcoupon.cpp

```diff
@@ -84,8 +84,34 @@ namespace QuantLib {
         return spanningTimeIndexMaturity_;
     }
 
-    Rate IborCoupon::indexFixing() const {
+    bool IborCoupon::hasFixed(Date refDate) const {
+        if (refDate == Date()) {
+            refDate = QuantLib::Settings::instance().evaluationDate();
+        }
+
+        if (fixingDate_ > refDate) {
+            return false;
+        } else if (fixingDate_ < refDate) {
+            return true;
+        } else {
+            // fixingDate_ == refDate
+            if (refDate == QuantLib::Settings::instance().evaluationDate()) {
+                if (QuantLib::Settings::instance().enforcesTodaysHistoricFixings()) {
+                    return true;
+                } else {
+                    try {
+                        return index_->pastFixing(fixingDate_) != Null<Real>();
+                    } catch (const Error&) {
+                        return false; // fall through and forecast
+                    }
+                }
+            } else {
+                return false;
```

Comment:
Here `fixingDate == refDate` but `refDate != evaluationDate`.  We should still check if there is a past fixing loaded.

I think we need a series of tests for this.

Reply:
I'm not entirely sure of what we should do in this particular case, though.

Reply:
Should the tests go in QuantLib/test-suite/cashflows.cpp? Or a new QuantLib/test-suite/iborcoupon.cpp?

Reply:
cashflows.cpp is ok.

Reply:
Hi Tom, it seems we stalled on this.  Do you have time to have another look?  Thanks!

Reply:
I would definitely like too. My current QuantLib priority is solving the mystery of the performance regression i see, which blocks any further upgrade. Once that's done i would like to get back to this, so we can stop carrying this patch. But of course, there will be other work to do at some point.